Various OSDK/OLM feats for 4.10 RN (#42246)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

[id="ocp-4-10-olm"]
=== Operator lifecycle

[id="ocp-4-10-copied-csvs"]
==== Disabling copied CSVs to support large clusters

> When an Operator is installed by Operator Lifecycle Manager (OLM), a simplified copy of its cluster service version (CSV) is created in every namespace that the Operator is configured to watch. These CSVs are known as copied CSVs and communicate to users which controllers are actively reconciling resource events in a given namespace.

Reviewer: Review ISG on "that" vs. "which" for restrictive clauses here. I'm not sure if that's even applicable here, as the phrasing of the final sentence sends my tired brain on a loop. :) 

Is it: "These CSVs are known as copied CSVs. They identify controllers that are actively reconciling resource events in a given namespace." ?

Author: The latter, yeah.

> On especially large clusters, with namespaces and installed Operators potentially in the hundreds or thousands, copied CSVs can consume an untenable amount of resources, such as OLM’s memory usage, cluster etcd limits, and networking. To support these larger clusters, cluster administrators can now disable copied CSVs for Operators installed with the `AllNamespaces` mode.

Reviewer: ```suggestion
On large clusters, with namespaces and installed Operators potentially in the hundreds or thousands, copied CSVs can consume an untenable amount of resources, such as OLM’s memory usage, cluster etcd limits, and networking <NOUN_HERE?>. To support these larger clusters, cluster administrators can now disable copied CSVs for Operators that are installed with the `AllNamespaces` mode.
```

Author: Went with "networking bandwidth".

For more details, see xref:../operators/admin/olm-config.adoc#olm-config[Configuring Operator Lifecycle Manager features].

[id="ocp-4-10-olm-hypershift"]
==== Operator Lifecycle Manager support for Hypershift

> Operator Lifecycle Manager (OLM) components, including Operator catalogs, can now run entirely on Hypershift-managed control planes. This capability does not occur any cost to tenants on worker nodes.

Reviewer: ```suggestion
Operator Lifecycle Manager (OLM) components, including Operator catalogs, can now run entirely on Hypershift-managed control planes. This capability does not incur any cost to tenants on worker nodes.
```
?

[id="ocp-4-10-olm-dependencies"]
==== Generic and complex constraints for dependencies

> Operators with specific dependency requirements can now use complex constraints or requirement expressions. The new `olm.constraint` bundle property holds dependency constraint information. A message field allows Operator authors to convey high-level details about why a particular constraint was used.

Reviewer: ```suggestion
Operators with specific dependency requirements can now use complex constraints or requirement expressions. The new `olm.constraint` bundle property holds dependency constraint information. A message field allows Operator authors to convey high-level details about why a particular constraint is used.
```

For more details, see _Operator Lifecycle Manager dependency resolution_.

[id="ocp-4-10-osdk"]
=== Operator development

[id="ocp-4-10-hybrid-helm-operator"]
==== Hybrid Helm Operator (Technology Preview)

The standard Helm-based Operator support in the Operator SDK has limited functionality compared to the Go-based and Ansible-based Operator support that has reached the Auto Pilot capability (level V) in the xref:../operators/understanding/olm-what-operators-are.adoc#olm-maturity-model_olm-what-operators-are[Operator maturity model].

> Starting in {product-title} 4.10 as a Technology Preview feature, the Operator SDK provides the Hybrid Helm Operator to enhance the existing Helm-based support's abilities through Go APIs. Operator authors can generate an Operator project beginning with a Helm chart, and then add advanced, event-based logic to the Helm reconciler in Go language. Authors can use Go to continue adding new APIs and custom resource definitions (CRDs) in the same project.

Reviewer: "provides" -> "includes" ? 

"support's" -> consider rewrite to eliminate possessive

For more details, see _Operator SDK tutorial for Hybrid Helm Operators_.

[id="ocp-4-10-ansible-operator-metrics"]
==== Custom metrics for Ansible-based Operators

> The Ansible-based Operator support in the Operator SDK now supports exposing custom metrics, emitting Kubernetes events, and better logging.

Reviewer: Could rewrite to eliminate duplication of "support."
